packages/ui/src/DropdownMenu/shared.ts

```diff
@@ -0,0 +1,63 @@
+import styled, { DefaultTheme } from 'styled-components';
+import { ActionType } from '../utils/ActionType.ts';
+
+export interface DropdownMenuItemBase {
+  actionType?: ActionType;
+  disabled?: boolean;
+}
+
+export interface StyledItemProps {
+  $actionType?: ActionType;
+  $disabled?: boolean;
+}
+
+const getColorByActionType = (theme: DefaultTheme, actionType: ActionType = 'default'): string => {
+  if (actionType === 'destructive') {
+    return theme.color.destructive.light;
+  }
+  return theme.color.text.primary;
+};
+
+const getOutlineColorByActionType = (
+  theme: DefaultTheme,
+  actionType: ActionType = 'default',
+): string => {
+  const map: Record<ActionType, keyof DefaultTheme['color']['action']> = {
+    default: 'neutralFocusOutline',
+    accent: 'primaryFocusOutline',
+    unshield: 'unshieldFocusOutline',
+    destructive: 'destructiveFocusOutline',
+  };
+  return theme.color.action[map[actionType]];
+};
+
+export const MenuItem = styled.div<StyledItemProps>`
+  display: flex;
+  align-items: center;
+  gap: ${props => props.theme.spacing(1)};
+  height: ${props => props.theme.spacing(8)};
+  padding: ${props => props.theme.spacing(1)} ${props => props.theme.spacing(2)};
+
+  border: none;
+  border-radius: ${props => props.theme.borderRadius.sm};
+
+  color: ${props => getColorByActionType(props.theme, props.$actionType)};
+  cursor: pointer;
+
+  background-color: transparent;
+  transition: background-color 0.15s;
+
+  &:focus:not(:disabled) {
+    background-color: ${props => props.theme.color.action.hoverOverlay};
+    outline: 2px solid ${props => getOutlineColorByActionType(props.theme, props.$actionType)};
+  }
```

Comment:
the styles do not have `:hover` and we can't change it – Radix does not distinguish between hover and focus for the menu items. It means that hovering an item also makes it focused to not confuse the states: https://github.com/radix-ui/primitives/issues/956#issuecomment-961819460